frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -390,6 +400,56 @@ function ReaderHighlight:addToMainMenu(menu_items)
             separator = i == #highlight_style,
         })
     end
+    if Screen:isColorScreen() then
+        table.insert(menu_items.highlight_options.sub_item_table, {
+            text_func = function()
+                local saved_color = self.view.highlight.saved_color or "yellow"
+                for __, v in ipairs(self.highlight_colors) do
+                    if v[2] == saved_color then
+                        return T(_("Highlight color: %1"), string.lower(v[1]))
+                    end
+                end
+                return T(_("Highlight color: %1"), saved_color)
+            end,
+            enabled_func = function()
+                return Screen:isColorEnabled() and self.view.highlight.saved_drawer ~= "invert"
+            end,
+            callback = function(touchmenu_instance)
+                local default_color = G_reader_settings:readSetting("highlight_color", "yellow")
+                local saved_color = self.view.highlight.saved_color or "yellow"
+                local radio_buttons = {}
+                for _, v in ipairs(self.highlight_colors) do
+                    table.insert(radio_buttons, {
+                        {
+                            text = v[1],
+                            checked = v[2] == saved_color,
+                            provider = v[2],
+                        },
+                    })
+                end
+                UIManager:show(require("ui/widget/radiobuttonwidget"):new{
+                    title_text = _("Highlight color"),
+                    width_factor = 0.5,
+                    keep_shown_on_apply = false,
+                    radio_buttons = radio_buttons,
+                    default_provider = default_color,
+                    callback = function(radio)
+                        self.view.highlight.saved_color = radio.provider
+                        self.ui.doc_settings:saveSetting("highlight_color", self.view.highlight.saved_color)
+                        UIManager:setDirty(self.dialog, "ui")
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                })
+            end,
+            hold_callback = function(touchmenu_instance)
+                G_reader_settings:saveSetting("highlight_color", self.view.highlight.saved_color)
+                UIManager:show(Notification:new{
+                    text = T(_("Default highlight color changed to '%1'."), self.view.highlight.saved_color),
```

Comment:
Reading this, it feels that with this, if saved_color is yellow, and I long-press on Red, it will save yellow as our global default? If yes, it feels a little odd :) I would store Red, even if the document default stays yellow.

Reply:
You can't long-press inside the RadioMenu popup. Doing so does nothing.
long-press only works on the "Highlight color: XYZ" menu item that opens the selector popup. As such it just saves the currently selected value without showing the selector.